### PR TITLE
hw-27.4

### PR DESCRIPTION
**Pull request by MaksymVakuliuk**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("SELECT DISTINCT o FROM Order o "
+                    + "LEFT JOIN FETCH o.tickets t "
+                    + "LEFT JOIN FETCH t.user "
+                    + "LEFT JOIN FETCH t.movieSession mS "
+                    + "LEFT JOIN FETCH mS.cinemaHall cH "
+                    + "LEFT JOIN FETCH mS.movie m "
+                    + "WHERE o.user = :user ");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:47]: try to do that only with left join fetch tickets.See if that gonna work)

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -37,10 +37,12 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
...
     public ShoppingCart getByUser(User user) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-
             Query<ShoppingCart> query = session
                     .createQuery("FROM ShoppingCart sc "
-                            + "LEFT JOIN FETCH sc.tickets shT"
+                            + "LEFT JOIN FETCH sc.tickets shT "
+                            + "LEFT JOIN FETCH shT.movieSession shTm "
+                            + "LEFT JOIN FETCH shTm.cinemaHall "
+                            + "LEFT JOIN FETCH shTm.movie "
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java:45]: same

File: src/main/java/com/dev/cinema/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+@Entity
+@Table(name = "\"order\"")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    private LocalDateTime orderDate;
+    @OneToOne(fetch = FetchType.LAZY)
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/model/Order.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY)
```

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -41,4 +42,11 @@ public void registerNewShoppingCart(User user) {
         shoppingCart.setUser(user);
         shoppingCartDao.add(shoppingCart);
     }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.setTickets(List.of());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        shoppingCart.setTickets(Collections.emptyList());
```
It gives more information about what exactly you are doing

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (MaksymVakuliuk) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:47]: @dmytro-hryhoruk Doesn't work LazyInitializationException.

**Review by dmytro-hryhoruk — COMMENTED** [review]
travis)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**